lib/parent/functional.ex

```diff
@@ -180,50 +181,43 @@ defmodule Parent.Functional do
   def shutdown_all(state, reason) do
     state = terminate_all_children(state, shutdown_reason(reason))
 
-    # The registry now contains only children which refused to die. These
-    # children are already forcefully terminated, so now we just have to
-    # wait for the :EXIT message.
-    Enum.each(Registry.entries(state.registry), fn {pid, _data} ->
-      receive do
-        {:EXIT, ^pid, _reason} -> :ok
-      end
-    end)
-
     %{state | registry: Registry.new()}
   end
 
   defp terminate_all_children(state, reason) do
     pids_and_specs =
       state.registry
       |> Registry.entries()
-      |> Enum.sort_by(fn {_pid, process} -> process.data.shutdown end)
-
-    # Kill all timeout timers first, because we don't want timeout to interfere with the shutdown logic.
-    Enum.each(pids_and_specs, fn {pid, process} -> kill_timer(process.data.timer_ref, pid) end)
+      |> Enum.sort_by(fn {_pid, process} -> process.data.startup_index end, :desc)
```

Comment:
Since we're using the new sort, we should add Elixir 1.10 as a minimum version in mix.exs.

Reply:
Good catch! To avoid bumping the Elixir version would you prefer if I didn't use the new API?

Reply:
I personally don't try too hard to keep compatibility with older Elixir versions, because that usually adds more complexity, and also because upgrading to the most recent Elixir is IME typically not very complicated. That said, making this work on pre 1.10 is trivial, so I'd slightly prefer that.

I wonder why CI failure isn't shown here. I can see the failure on the travis site, but don't see anything here. I probably need to configure something. I'll see if I can make that work.